src/controller/CHIPDeviceController.cpp

```diff
@@ -427,5 +427,14 @@ void ChipDeviceController::OnRendezvousMessageReceived(PacketBuffer * buffer)
     }
 }
 
+void ChipDeviceController::OnRendezvousStatusUpdate(RendezvousSessionDelegate::Status status)
+{
+    if (status == RendezvousSessionDelegate::NetworkProvisioningSuccess && mRendezvousSession->HasIPAddress())
```

Comment:
I guess it does not hurt to double check but can we really have `status == RendezvousSessionDelegate::NetworkProvisioningSuccess` and `mRendezvousSession->HasIPAddress()` to be false ?
Seems redundant.